src/bloqade/squin/noise/stmts.py

```diff
@@ -49,11 +49,18 @@ class PauliChannel(NoiseChannel):
     # TODO add validation for params (maybe during lowering via custom lower?)
     n_qubits: int = info.attribute()
     params: ir.SSAValue = info.argument(types.Tuple[types.Vararg(types.Float)])
-    result: ir.ResultValue = info.result(OpType)
 
 
 @statement(dialect=dialect)
 class QubitLoss(NoiseChannel):
     # NOTE: qubit loss error (not supported by Stim)
     p: ir.SSAValue = info.argument(types.Float)
+
+
+@statement(dialect=dialect)
+class StochasticUnitaryChannel(ir.Statement):
+    operators: ir.SSAValue = info.argument(ilist.IListType[OpType, NumOperators])
```

Comment:
One issue here is what happens when the operators do not have the same number of sites. And also does this mean we might want to make `OpType` Generic with a type parameter for the operator size?

cc: @Roger-luo @johnzl-777 @kaihsin

Reply:
Well given the current noise statements, I don't think there can be a case where the number of sites is different. I could check this on runtime and throw an error if needed.

If we'd want to accommodate different sizes here, I think we can't have this share the same `OperatorRuntime`, since these all have a fixed number of sites. Also, what would be the use case?

> also what does this statement mean? randomly apply one of the unitary? What is the use case?

We're rewriting a bunch of noise statements to it so we can have a simple runtime. The noise statements are all represented by this single statement.

Reply:
> I could check this on runtime and throw an error if needed.

Yeah we should check it at runtime in this case, we can add an attribute to say it's validated to turn the checks off if we have validation later.

> We're rewriting a bunch of noise statements to it so we can have a simple runtime. The noise statements are all represented by this single statement.

I see! make sense!